h2o-r/h2o-package/R/models.R

```diff
@@ -1361,6 +1361,116 @@ h2o.thresholds_and_metric_scores <- function(object, train=FALSE, valid=FALSE) {
     invisible(NULL)
 }
 
+#' Retrieve the default Qini value
+#'
+#' Retrieves the Qini value from an \linkS4class{H2OBinomialUpliftMetrics}. If you need specific AECU value, see h2o.aecu_table function.
+#' If "train" and "valid" parameters are FALSE (default), then the training Qini value is returned. If more
+#' than one parameter is set to TRUE, then a named vector of Qini values are returned, where the names are "train", "valid".
+#'
+#' @param object An \linkS4class{H2OBinomialUpliftMetrics}
+#' @param train Retrieve the training Qini value
+#' @param valid Retrieve the validation Qini
+#' @examples
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' f <- "https://s3.amazonaws.com/h2o-public-test-data/smalldata/uplift/criteo_uplift_13k.csv"
+#' train <- h2o.importFile(f)
+#' train$treatment <- as.factor(train$treatment)
+#' train$conversion <- as.factor(train$conversion)
+#' 
+#' model <- h2o.upliftRandomForest(training_frame=train, x=sprintf("f%s",seq(0:10)), y="conversion",
+#'                                        ntrees=10, max_depth=5, treatment_column="treatment", 
+#'                                        auuc_type="AUTO")
+#' perf <- h2o.performance(model, train=TRUE) 
+#' h2o.qini(perf)
+#' }
+#' @export
+h2o.qini <- function(object, train=FALSE, valid=FALSE) {
+    if( is(object, "H2OModelMetrics") ) return( object@metrics$qini )
+    if( is(object, "H2OModel") ) {
+        model.parts <- .model.parts(object)
+        if ( !train && !valid ) {
+            metric <- model.parts$tm@metrics$qini
+            if ( !is.null(metric) ) return(metric)
+        }
+        v <- c()
+        v_names <- c()
+        if ( train ) {
+            v <- c(v,model.parts$tm@metrics$qini)
+            v_names <- c(v_names,"train")
+        }
+        if ( valid ) {
+            if( is.null(model.parts$vm) ) return(invisible(.warn.no.validation()))
+            else {
+                v <- c(v,model.parts$vm@metrics$qini)
+                v_names <- c(v_names,"valid")
+            }
+        }
+        if ( !is.null(v) ) {
+            names(v) <- v_names
+            if ( length(v)==1 ) { return( v[[1]] ) } else { return( v ) }
+        }
+    }
+    warning(paste0("No Qini value for ", class(object)))
+    invisible(NULL)
+}
+
```

Comment:
We don't implement the `h2o.aecu()` equivalent for pythons `def aecu(self, metric="qini"):`?